gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java

```diff
@@ -1605,6 +1647,399 @@ public Operation deleteNetwork(String network, OperationOption... options) {
     return deleteNetwork(NetworkId.of(network));
   }
 
+  @Override
+  public Operation create(InstanceInfo instance, OperationOption... options) {
+    final InstanceInfo completeInstance = instance.setProjectId(options().projectId());
+    final Map<ComputeRpc.Option, ?> optionsMap = optionMap(options);
+    try {
+      com.google.api.services.compute.model.Operation answer =
+          runWithRetries(new Callable<com.google.api.services.compute.model.Operation>() {
+            @Override
+            public com.google.api.services.compute.model.Operation call() {
+              return computeRpc.createInstance(completeInstance.instanceId().zone(),
+                  completeInstance.toPb(), optionsMap);
+            }
+          }, options().retryParams(), EXCEPTION_HANDLER);
+      return answer == null ? null : Operation.fromPb(this, answer);
+    } catch (RetryHelper.RetryHelperException e) {
+      throw ComputeException.translateAndThrow(e);
+    }
+  }
+
+  @Override
+  public Instance get(final InstanceId instance, InstanceOption... options) {
+    final Map<ComputeRpc.Option, ?> optionsMap = optionMap(options);
+    try {
+      com.google.api.services.compute.model.Instance answer =
+          runWithRetries(new Callable<com.google.api.services.compute.model.Instance>() {
+            @Override
+            public com.google.api.services.compute.model.Instance call() {
+              return computeRpc.getInstance(instance.zone(), instance.instance(), optionsMap);
+            }
+          }, options().retryParams(), EXCEPTION_HANDLER);
+      return answer == null ? null : Instance.fromPb(this, answer);
+    } catch (RetryHelper.RetryHelperException e) {
+      throw ComputeException.translateAndThrow(e);
+    }
+  }
+
+  private static Function<com.google.api.services.compute.model.Instance, Instance>
+      instanceFromPb(final ComputeOptions serviceOptions) {
```

Comment:
This is a factory method for conversion functions. You pass the service options and this returns a `Function` object that does exactly what `Instance.fromPb` does. We need this to create `Function` objects for conversion to pass to Guava's `Iterables.transform`.